pkg/controller/volume/attachdetach/statusupdater/node_status_updater.go

```diff
@@ -69,10 +69,10 @@ func (nsu *nodeStatusUpdater) UpdateNodeStatuses() error {
 			// reset flag statusUpdateNeeded back to true to indicate this node status
 			// needs to be updated again
 			glog.V(2).Infof(
-				"Could not update node status. Failed to find node %q in NodeInformer cache. %v",
+				"Could not update node status. Failed to find node %q in NodeInformer cache. Error: '%v'",
 				nodeName,
 				err)
-			nsu.actualStateOfWorld.SetNodeStatusUpdateNeeded(nodeName)
+			nsu.actualStateOfWorld.RemoveNodeFromAttachUpdates(nodeName)
```

Comment:
The error is different if node does not exist errors.NewNotFound(v1.Resource("node"), name)
you can check the error type to determine whether it is because of node exist or not.